### PR TITLE
Implement emergency shutdown

### PR DESCRIPTION
**Pull request by haythem96**
# Implement emergency shutdown

## High Level Description

This PR implement the emergency shutdown mechanism along with the pausing mechanism.

- Added another `pauser` address that can pause the system. (pausing system = pausing all functionalities other than exercise and settling vault)
- Terminator can shutdown the system (stop all functionalities)

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [ ] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

**Review by antoncoding — APPROVED** [review]
LGTM

File: contracts/Controller.sol [hunk 1]
```diff
@@ -267,7 +329,7 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
      * @dev can only be called when system is not paused
      * @param _actions array of actions arguments
      */
-    function operate(Actions.ActionArgs[] memory _actions) external payable nonReentrant {
+    function operate(Actions.ActionArgs[] memory _actions) external payable nonReentrant notShutdown {
```
Comment (aparnakr) [inline contracts/Controller.sol:332]: Just to verify my understanding:

1. If the system is paused, mint, short, burn, add collateral (all actions except exercise / settleVault are paused
2. if the system is shutdown, all actions are stopped

Comment (haythem96) [reply contracts/Controller.sol:332]: Yeah right.

File: test/controller.test.ts [hunk 2]
```diff
@@ -3012,50 +3734,112 @@ contract('Controller', ([owner, accountOwner1, accountOwner2, accountOperator1,
...
-        // whitelist otoken to be used in the protocol
-        await whitelist.whitelistOtoken(secondShortOtoken.address, {from: owner})
-        // open new vault, mint naked short, sell it to holder 1
-        collateralToDespoit = new BigNumber(await secondShortOtoken.strikePrice()).dividedBy(1e18)
-        amountToMint = new BigNumber('1')
-        vaultCounter = new BigNumber(await controllerProxy.getAccountVaultCounter(accountOwner1)).plus(1)
-        actionArgs = [
+      })
+
+      it('should set pauser address', async () => {
```
Comment (aparnakr) [inline test/controller.test.ts:3754]:
```suggestion
      it('should set terminator address', async () => {
```

File: test/controller.test.ts [hunk 3]
```diff
@@ -3012,50 +3734,112 @@ contract('Controller', ([owner, accountOwner1, accountOwner2, accountOperator1,
...
+        await controllerProxy.setTerminator(terminator, {from: owner})
+        assert.equal(await controllerProxy.terminator(), terminator, 'Terminator address mismatch')
+      })
+
+      it('should revert when trigerring emergency shutdown the system from address other than pauser', async () => {
```
Comment (aparnakr) [inline test/controller.test.ts:3759]:
```suggestion
      it('should revert when trigerring emergency shutdown the system from address other than terminator', async () => {
```

**Review by aparnakr — APPROVED** [review]
LGTM!

**Pull request merged**